scripts/dns-test-execute.bash

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# The integration tests call this to test Linux DNS control, using the `/etc/resolv.conf`
+# method which only works well inside Alpine Docker containers.
+
+set -euo pipefail
+
+HTTPBIN=test.httpbin.docker.local
+
+# Wait for client to ping httpbin (CIDR) resource through the gateway
+docker compose exec -it client timeout 60 \
+sh -c "until ping -W 1 -c 10 172.20.0.100 &>/dev/null; do true; done"
+
+echo "# check original resolv.conf"
+docker compose exec -it client sh -c "cat /etc/resolv.conf.firezone-backup"
+
+echo "# Make sure gateway can reach httpbin by DNS"
+docker compose exec -it gateway sh -c "curl $HTTPBIN/get"
```

Comment:
Would it make sense to write this as, kind of as an assertion?

```suggestion
docker compose exec -it gateway sh -c "curl $HTTPBIN/get" || echo "gateway cannot reach httpbin by DNS"
```

I don't know if bash will forward the exit code correctly in this case but I think it will.

Reply:
Also, don't you need to run `curl` with `--exit-code` or something to fail if it is not reachable?

Reply:
I'll check on `--exit-code`. I had changed the comments into unconditional echoes so that the output was easier to debug and I was thinking of just leaving them in. It does violate the "Errors should be noisy, success should be silent" principle, but it's a debug log that would only be read if the test fails or needs to be re-validated.

It's also similar to, in the fully-YAML tests, they're split up into many small steps. But if we shell out to one big Bash script, the distinction between the steps is lost

Reply:
I think for connection issues curl already returns non-zero, but I added `--fail` just to be sure. I think that only changes a 404 or 500 error from 0 to non-zero

Reply:
> I had changed the comments into unconditional echoes so that the output was easier to debug and I was thinking of just leaving them in.

I find `set -x` to be quite useful for that.